internal/completions/asdf.zsh

```diff
@@ -92,110 +101,246 @@ _asdf__installed_versions_of_plus_system() {
           expl "Latest version" \
           compadd -- latest:${^$(asdf list all "$tool")}
-                else
-                  _wanted "latest-tag-$tool" \
-                    expl "Latest version" \
-                    compadd -- 'latest' 'latest:'
-                                      _wanted "remote-versions-$tool" \
-                                        expl "Available versions of $tool" \
-                                        compadd -- $(asdf list all "$tool")
+      else
+        # Offer both latest options and specific versions
+        _wanted "latest-tag-$tool" \
+          expl "Latest version" \
+          compadd -- 'latest' 'latest:'
+        _wanted "remote-versions-$tool" \
+          expl "Available versions of $tool" \
+          compadd -- $(asdf list all "$tool")
       fi
       return
     fi
     ;;
   (latest)
+    # Complete plugin names or --all flag for latest command
     if (( CURRENT == 3 )); then
       _alternative  \
-        'all:all:(--all)' \
+        'flags:flags:((--all\:"Show latest version of all tools"))' \
         'asdf-available-plugins:Installed ASDF Plugins:_asdf__installed_plugins'
-            elif (( CURRENT == 4)); then
-              local tool="${words[3]}"
-              local query=${words[4]}
-              [[ -n $query ]] || query='[0-9]'
-              _wanted "latest-pattern-$tool" \
-                expl "Pattern to look for in matching versions of $tool" \
-                compadd -- $(asdf list all "$tool" "$query")
     fi
     ;;
-  (uninstall|reshim)
-    compset -n 2
-    _arguments '1:plugin-name: _asdf__installed_plugins' '2:tool-version:{_asdf__installed_versions_of ${words[2]}}'
+  (uninstall|reshim|where)
+    # Handle complex install command completion with latest tag support
+    if (( CURRENT == 3 )); then
+      _asdf__installed_plugins
+      return
+    elif (( CURRENT == 4 )); then
+      # For normal list: show installed versions with optional filter
+      _asdf__installed_versions_of ${words[3]}
+      return
+    fi
     ;;
   (set)
-    compset -n 2
-    _arguments '1:plugin-name: _asdf__installed_plugins' '2:tool-version:{_asdf__installed_versions_of_plus_system ${words[2]}}'
-    ;;
-  (where)
-    # version is optional
-    compset -n 2
-    _arguments '1:plugin-name: _asdf__installed_plugins' '2::tool-version:{_asdf__installed_versions_of ${words[2]}}'
+    # Handle set command completion
+    case $CURRENT in
+      3)
+        _alternative \
+          'flags:flags:((-u\:"set version in user home directory" -p\:"set version in closest parent .tool-versions"))' \
+          'plugin:plugin:_asdf__installed_plugins'
+        ;;
+      4)
+        if [[ ${words[3]} == -* ]]; then
+          # After flag, complete with plugin name
+          _asdf__installed_plugins
+        else
+          # Complete with available versions for the plugin
+          local versions
+          if versions=$(asdf list all "${words[3]}" 2>/dev/null); then
```

Comment:
Before the change this listed only the versions that were installed, after this change, it lists everything (same as line 330). See https://github.com/asdf-vm/asdf/issues/2134